pkg/envoy/envoy_container_logs.go

```diff
@@ -31,49 +29,12 @@ func HasNoBadEnvoyLogsCheck(client kubernetes.Interface, pod *v1.Pod) NoBadEnvoy
 
 // Info implements common.Runnable
 func (check NoBadEnvoyLogsCheck) Info() string {
-	return fmt.Sprintf("Checking whether pod %s has bad/error logs in envoy container", check.pod.Name)
+	return fmt.Sprintf("Checking whether pod %s has bad (fatal/error/warning/fail) logs in envoy container", check.pod.Name)
 }
 
 // Run implements common.Runnable
 func (check NoBadEnvoyLogsCheck) Run() error {
-	envoyLogsTailLines := int64(10)
-	envoyContainerName := "envoy"
-	podLogsOpt := v1.PodLogOptions{
-		Container: envoyContainerName,
-		Follow:    false,
-		Previous:  false,
-		TailLines: &envoyLogsTailLines,
-	}
-
-	request := check.client.CoreV1().Pods(check.pod.Namespace).GetLogs(check.pod.Name, &podLogsOpt)
-	envoyPodLogsReader, err := request.Stream(context.TODO())
-	if err != nil {
-		// If there are issues obtaining current container logs, return previously terminated container logs.
-		podLogsOpt.Previous = true
-		request := check.client.CoreV1().Pods(check.pod.Namespace).GetLogs(check.pod.Name, &podLogsOpt)
-		envoyPodLogsReader, err = request.Stream(context.TODO())
-		if err != nil {
-			return fmt.Errorf("could not obtain %s container logs of pod %s: %#v", envoyContainerName, check.pod.Name, err)
-		}
-	}
-	defer envoyPodLogsReader.Close() //nolint: errcheck,gosec
-
-	re := regexp.MustCompile("(?i)(error)|(warn)")
-	scanner := bufio.NewScanner(envoyPodLogsReader)
-	var badEnvoyLogLines string
-	for scanner.Scan() {
-		logLine := scanner.Text()
-		if re.MatchString(logLine) {
-			badEnvoyLogLines += logLine + "\n"
-		}
-	}
-
-	if len(badEnvoyLogLines) != 0 {
-		log.Error().Msgf("%s container of pod %s contains bad logs", envoyContainerName, check.pod.Name)
-		log.Error().Msg(badEnvoyLogLines)
-	}
-
-	return nil
+	return podhelper.HasNoBadLogs(check.client, check.pod, "envoy")
```

Comment:
@draychev Regarding your comment https://github.com/openservicemesh/osm-health/pull/50/files#r683754375

Both `osm-health connectivity pod-to-pod srcPod destPod` and the `osm-health control-plane status` now use `podhelper.HasNoBadLogs`.

See the following code:

(This one, checking envoy container logs, is invoked as part of pod to pod)
https://github.com/openservicemesh/osm-health/pull/50/files#diff-533a363c43aa9facdfece27a0c8e9d31a8c22ec541dd4d172d6c782d10eb9c68R37

(This one is used in the control-plane status check for osm-controller log issues)
https://github.com/openservicemesh/osm-health/pull/50/files#diff-d4f2e551640a295a560017296694c7c15fb6f0c302dfd34c013894ddb0c893ddR51-R57